### PR TITLE
faster yaml loading

### PR DESCRIPTION
**Pull request by ilaner**
## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-7455

## Description
Don't use round trip YAML loading in content parser and in `find_type`

## Nightly
https://code.pan.run/xsoar/content/-/pipelines/5785943

### CODE REVIEW

File: demisto_sdk/commands/common/tools.py [hunk 1]
```diff
@@ -539,16 +540,19 @@ def get_remote_file_from_api(
...
     file_content,
     full_file_path: str,
+    keep_order: bool = False,
 ) -> Dict:
     if full_file_path.endswith("json"):
         file_details = json.loads(file_content)
     elif full_file_path.endswith(("yml", "yaml")):
+        if keep_order:
+            file_details = yaml_ordered_load.load(file_content)
         file_details = yaml.load(file_content)
```
Comment (dorschw) [inline demisto_sdk/commands/common/tools.py:556]:
```suggestion
        file_details = yaml_ordered_load.load(file_content) if keep_order else yaml.load(file_content)
```

Comment (dorschw) [reply demisto_sdk/commands/common/tools.py:556]: or, the chaotic equivalent (don't)

```suggestion
        file_details = (yaml_ordered_load if keep_order else yaml).load(file_content)
```

File: demisto_sdk/commands/format/update_generic.py [hunk 2]
```diff
@@ -379,7 +379,7 @@ def set_from_version_key_name(self) -> Union[str, None]:
     def is_old_file(path: str, prev_ver: str) -> dict:
         """Check whether the file is in git repo or new file."""
         if path:
-            data = get_remote_file(path, prev_ver)
+            data = get_remote_file(path, prev_ver, keep_order=True)
```
Comment (dorschw) [inline demisto_sdk/commands/format/update_generic.py:382]:
```suggestion
            data = get_remote_file(path, prev_ver, keep_order=True) # keep_order=True as format matches the order to the previous version
```

**Review by dorschw — APPROVED** [review]

File: demisto_sdk/commands/common/tools.py [hunk 3]
```diff
@@ -809,6 +809,7 @@ def get_file(
     type_of_file: Optional[str] = None,
     clear_cache: bool = False,
     return_content: bool = False,
+    keep_order: bool = True,
```
Comment (dorschw) [inline demisto_sdk/commands/common/tools.py:812]: why default=`True`? It's slower and doesn't matter in almost all cases

Comment (ilaner) [reply demisto_sdk/commands/common/tools.py:812]: because many UT fail of it

Comment (dorschw) [reply demisto_sdk/commands/common/tools.py:812]: Let's open a task for this, then a followup task to fix the UTs

Comment (ilaner) [reply demisto_sdk/commands/common/tools.py:812]: OK, I'll open now

File: demisto_sdk/commands/common/handlers/yaml/ruamel_handler.py [hunk 4]
```diff
@@ -14,11 +14,20 @@ class RUAMEL_Handler(XSOAR_Handler):
...
         preserve_quotes=True,
         allow_duplicate_keys=False,
         width=5000,
         ensure_ascii=False,
     ):
+        """
+        typ: 'rt'/None -> RoundTripLoader/RoundTripDumper,  (default)
+             'safe'    -> SafeLoader/SafeDumper,
+             'unsafe'  -> normal/unsafe Loader/Dumper
+             'base'    -> baseloader
```
Comment (dorschw) [inline demisto_sdk/commands/common/handlers/yaml/ruamel_handler.py:27]:
```suggestion
        typ: 'rt'/None -> RoundTripLoader/RoundTripDumper,  (default, supports comments, slower then the rest)
             'safe'    -> SafeLoader/SafeDumper,
             'unsafe'  -> normal/unsafe Loader/Dumper
             'base'    -> baseloader
```

**Review by dorschw — APPROVED** [review]

**Pull request merged**